consumer_monitor.go

```diff
@@ -0,0 +1,266 @@
+package kafka
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Financial-Times/go-logger/v2"
+	"github.com/Shopify/sarama"
+)
+
+type offsetFetcher interface {
+	ListConsumerGroupOffsets(group string, topicPartitions map[string][]int32) (*sarama.OffsetFetchResponse, error)
+
+	Close() error
+}
+
+type fetcherScheduler struct {
+	ticker            *time.Ticker
+	standardInterval  time.Duration
+	shortenedInterval time.Duration
+	failureCount      int
+	maxFailureCount   int
+}
+
+type consumerMonitor struct {
+	consumerGroup string
+	offsetFetcher offsetFetcher
+	scheduler     fetcherScheduler
+	// Key1 is Topic. Key2 is Partition. Value is Offset.
+	subscriptions map[string]map[int32]int64
+	topicsLock    *sync.RWMutex
+	topics        []*Topic
+	unknownStatus bool
+	logger        *logger.UPPLogger
+}
+
+func newConsumerMonitor(config ConsumerConfig, fetcher offsetFetcher, topics []*Topic, logger *logger.UPPLogger) *consumerMonitor {
+	offsetFetchInterval := 3 * time.Minute
+	if config.OffsetFetchInterval != 0 {
```

Comment:
What is the behavior of consumerMonitor going to be if we pass a negative number or a very large number here? Should we define value boundaries and add some validation? 🤔 
Same question goes for maxFailureCount